arch/arm/src/common/Toolchain.defs

```diff
@@ -220,9 +226,10 @@ ifneq ($(CONFIG_CXX_RTTI),y)
 endif
 
 ARCHOPTIMIZATION += -fno-common -Wall -Wshadow -Wundef
-ARCHOPTIMIZATION += -nostdlib
 
-ifeq ($(CONFIG_ARM_TOOLCHAIN_ARMCLANG),)
+ifeq ($(CONFIG_ARM_TOOLCHAIN_ARMCLANG),y)
+  ARCHOPTIMIZATION += -nostdlib
```

Comment:
So till now this flag was added for all cases, but now only in case of ARM CLANG?

Reply:
Yes, this is a regression issue, PR https://github.com/apache/incubator-nuttx/pull/7100 added this option but non-armclang toolchains need to use the standard system startup files or libraries, this change will broken GCC compiler if enable -flto.